tests/cpp/operator/mkldnn.cc

```diff
@@ -1204,6 +1271,155 @@ void TestPoolingOp(const OpAttrs &forward_attrs, const OpAttrs &backwards_attrs)
+        != in_arr.arr.shape().ndim())
+      continue;
+
+    float scale = CalculateWidthConvOutput(input_shape[2], kernel[0], padding[0], stride[0])
+        / static_cast<float>(input_shape[2]);
+
+    if (is_deconv) {
+      scale = CalculateWidthDeconvOutput(input_shape[2], kernel[0], padding[0], stride[0])
+          / static_cast<float>(input_shape[2]);
+    }
+    std::vector<float> scale_vector(in_arr.arr.shape().ndim());
+    scale_vector[0] = 1;
+    scale_vector[1] = static_cast<float>(num_filter) / input_shape[1];
+    scale_vector[2] = scale;
+    scale_vector[3] = scale;
+
+    for (int i = 0; i < forward_attrs.num_outputs; i++) {
+      out_arrs[i] = GetTestOutputArrays(in_arr.arr.shape(), pds, scale_vector);
+      ex_out_arrs[i] = GetTestOutputArrays(in_arr.arr.shape(), pds, scale_vector);
+    }
+    NDArray ndkernel = CreateKernelNDArray(kernel, num_filter, in_arr.arr.shape(), is_deconv);
+    NDArray ndbias = CreateBiasNDArray(num_filter);
+    inputs[0] = &in_arr.arr;
+    inputs[1] = &ndkernel;
+    inputs[2] = &ndbias;
+    for (size_t output_i = 0; output_i < out_arrs[0].size(); output_i++) {
+      for (int i = 0; i < forward_attrs.num_outputs; i++) {
+        req[i] = kWriteTo;
+        outputs[i] = &out_arrs[i][output_i].arr;
+        ex_outputs[i] = &ex_out_arrs[i][output_i].arr;
+      }
+      Imperative::Get()->set_is_training(true);
+
+      PrintVerifyMsg(in_arr, out_arrs[0][output_i]);
+      Imperative::Get()->InvokeOp(Context(), forward_attrs.attrs, inputs,
+                                  outputs, req, DispatchMode::kFCompute, mxnet::OpStatePtr());
+      Imperative::Get()->InvokeOp(Context(), forward_attrs.attrs, inputs,
+                                  ex_outputs, req, DispatchMode::kFComputeEx, mxnet::OpStatePtr());
+      Engine::Get()->WaitForAll();
+      VerifyCopyResult(outputs, ex_outputs);
+
+      // backwards test performed same time since output needed
+      backwards_input[0] = outputs[0];  // output grad
+      backwards_input[1] = inputs[0];  // input
+      backwards_input[2] = inputs[1];  // kernel
+      backwards_input[3] = inputs[2];  // bias
+
+      auto tmp_output = GetTestInputArrays(true)[i1];
+      NDArray tmp_kernel = CreateKernelNDArray(kernel, num_filter, in_arr.arr.shape(), is_deconv);
+      NDArray tmp_bias = CreateBiasNDArray(num_filter);
+
+      backwards_outputs[0] = &tmp_output.arr;
+      backwards_outputs[1] = &tmp_kernel;
+      backwards_outputs[2] = &tmp_bias;
+
+      auto tmp_output2 = GetTestInputArrays(true)[i1];
+      NDArray tmp_kernel2 = CreateKernelNDArray(kernel, num_filter, in_arr.arr.shape(), is_deconv);
+      NDArray tmp_bias2 = CreateBiasNDArray(num_filter);
+      backwards_ex_outputs[0] = &tmp_output2.arr;
+      backwards_ex_outputs[1] = &tmp_kernel2;
+      backwards_ex_outputs[2] = &tmp_bias2;
+
+      for (int i = 0; i < backwards_attrs.num_outputs; i++)
+        back_req[0] = kWriteTo;
```

Comment:
only WriteTo? I think you should test AddTo as well.

Reply:
regular fcompute for deconvolution/convolution does not support kAddTo currently
https://github.com/azai91/incubator-mxnet/blob/d79e1ad3294837cac653478045023fd312ceed78/src/operator/nn/convolution-inl.h#L178

https://github.com/azai91/incubator-mxnet/blob/46e47cbc6183d2812a2e405851f0b209383e72ad/src/operator/nn/deconvolution-inl.h#L223